lib/src/utils.rs

```diff
@@ -63,30 +60,31 @@ pub fn generate_transactions(
     anchor_tx: Option<AlloyTransaction>,
 ) -> Vec<TxEnvelope> {
     // Decode the tx list from the raw data posted onchain
-    let tx_list = &if is_blob_data {
-        let compressed_tx_list = decode_blob_data(tx_list);
-        zlib_decompress_data(&compressed_tx_list).unwrap_or_default()
-    } else {
-        if chain_spec.network() == Some(Network::TaikoA7) {
+    let tx_list = match (
+        is_blob_data,
+        chain_spec.network(),
+        validate_calldata_tx_list(tx_list),
+    ) {
```

Comment:
Call me old fashioned, but I'm not a big fan of this one. Probably because I'm much less used to this kind of syntax, but for me the if/else way is much easier to understand what's going on.